### PR TITLE
replace `argonautica` with `rust-argon2`

### PR DESCRIPTION
**Pull request by XBagon**
`argonautica` seems to be unmaintained for 3 years now and only supports the C-implementation. `rust-argon2` seems to be the go-to modern pure-rust implementation, so I tried to replace it bit for bit as well as I could.

### CODE REVIEW

**Comment by Wulf** [comment]
hey @XBagon, thanks a lot for this PR -- I was actually going to look into this! Sorry for reviewing this so late, it's been a busy few weeks.

**Review by Wulf — CHANGES_REQUESTED** [review]
Just a minor comment, let me know what you think!

File: create-rust-app/src/auth/controller.rs [hunk 1]
```diff
@@ -713,3 +687,10 @@ pub fn reset_password(
 
     Ok(())
 }
+
+pub fn generate_salt() -> [u8; 8] {
+    use rand::Fill;
+    let mut salt = [0; 8];
+    salt.try_fill(&mut rand::thread_rng()).unwrap();
+    salt
+}
```
[the remark this answers is not in the document]
Comment (XBagon) [reply create-rust-app/src/auth/controller.rs:696]: I personally have no problem with this change, just did some minor reasearch to understand which improvements that would bring for security, as it seems probably none.
https://stackoverflow.com/a/5197921

[the remark this answers is not in the document]
Comment (Wulf) [reply create-rust-app/src/auth/controller.rs:696]: thanks for sharing @XBagon. I agree, after reading more about it, it seems to be almost negligible. Let's use 16 bytes as recommended in the [argon2 paper](https://raw.githubusercontent.com/P-H-C/phc-winner-argon2/master/argon2-specs.pdf).

image

[the remark this answers is not in the document]
Comment (Wulf) [reply create-rust-app/src/auth/controller.rs:696]: Also, I also want to say sorry for dragging this out for long -- I had to make sure we did the right thing :)

**Comment by Wulf** [comment]
I'm going to merge this to the `rust-argon2` branch and create a new PR (I can't figure out how to create a PR for your repo -- it says I already have the repository forked :sweat_smile:)

**Comment by XBagon** [comment]
> I'm going to merge this to the `rust-argon2` branch and create a new PR (I can't figure out how to create a PR for your repo -- it says I already have the repository forked 😅)

Okay, sounds like weird behaviour..
Thanks for finishing it, didn't get around to do the small change 😅

**Pull request merged**